service/worker/scanner/executions/task.go

```diff
@@ -167,6 +181,11 @@ func (t *task) validate(
 		results = append(results, validationResults...)
 	}
 
+	// Fail fast if the mutable is corrupted, no need to validate history.
+	if len(results) > 0 {
```

Comment:
shall we change the mutableStateValidator.Validate() logic to check retention first and if it exceeds retention it should not do any further checking. Also we should enable delete when it detect closed executions exceeds retention.

Reply:
Yes. Currently, it will delete if expired data detected.

Reply:
Update to return if the retention check fails.